### PR TITLE
 vmemcache_evict: error out if repl_p_lru_evict fails

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/167)

### CODE REVIEW

**Comment by ldorau** [comment]
Rebased

File: src/libvmemcache.h [hunk 1]
```diff
@@ -80,7 +80,7 @@ extern "C" {
  */
 typedef struct vmemcache VMEMcache;
 
-enum vmemcache_replacement_policy {
+enum vmemcache_repl_p {
```
Comment (marcinslusarz) [inline src/libvmemcache.h]: rationale? you are changing public interface...

Comment (kilobyte) [reply src/libvmemcache.h]: Heh, I'm rewriting the oonfig API, and also named the field `repl_p`.  Łukasz's change seems natural — Cobol Fingers is a nasty disease, let's protect users against it :)

Comment (ldorau) [reply src/libvmemcache.h]: It is shorter and this name ("repl_p") is already used in vmemcache_repl.c.
The library has not been released yet, so we can change the public interface.

**Comment by marcinslusarz** [comment]
tests?

**Comment by ldorau** [comment]
Rebased

**Comment by ldorau** [comment]
> tests?

It is just a fix for an error path - it did not fail, where it should to. I cannot make up a test for that.

**Comment by marcinslusarz** [comment]
All tests pass both before and after the fix, so how can I evaluate this PR (or its revert when it turns out to be wrong) without a test that verifies the fix?

**Comment by ldorau** [comment]
I will try to make up a test for this case

**Comment by ldorau** [comment]
@marcinslusarz a test was added and the fix was fixed ;-)

Since now this PR requires https://github.com/pmem/vmemcache/pull/191 to be merged first.

**Comment by ldorau** [comment]
> Since now this PR requires #191 to be merged first.
>
... in order that the test failed without the fix.

**Comment by ldorau** [comment]
@marcinslusarz This the 3rd PR to be merged.

**Comment by ldorau** [comment]
??? Please read the code carefully - there is ONLY ONE thread that is evicting...

**Comment by ldorau** [comment]
This test simulates such extremely difficult conditions for an eviction that even only one can be too many...

**Comment by ldorau** [comment]
> Ah crap, you're right. But in that case, why won't we demand that **all** evicts succeed?
> 

I added the comment:
```
	 * It is very likely that even all of these vmemcache_evict() calls
	 * will fail, because the conditions are extremely difficult (all cache
	 * entries are being constantly read (used) by separate threads),
	 * but this is acceptable, because this test is dedicated
	 * to test the failure path of vmemcache_evict()
	 * and the success criteria of this test are checks done in free_cache()
	 * at the end of the test.
```

**Pull request merged**